components/do_verify_email/do_verify_email.tsx

```diff
@@ -1,104 +1,88 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {Error} from 'mattermost-redux/types/errors';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {ActionFunc, ActionResult} from 'mattermost-redux/types/actions';
+
 import {trackEvent} from 'actions/diagnostics_actions.jsx';
 import {browserHistory} from 'utils/browser_history';
-import {AnnouncementBarTypes, AnnouncementBarMessages, VerifyEmailErrors} from 'utils/constants';
+import {AnnouncementBarMessages, VerifyEmailErrors} from 'utils/constants';
 import logoImage from 'images/logo.png';
 import BackButton from 'components/common/back_button';
 import LoadingScreen from 'components/loading_screen';
 
 import * as GlobalActions from 'actions/global_actions.jsx';
 
-export default class DoVerifyEmail extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Object with validation parameters given in link
-         */
-        location: PropTypes.object.isRequired,
-
-        /**
-         * Title of the app or site.
-         */
-        siteName: PropTypes.string,
-
-        /*
-         * Object with redux action creators
-         */
-        actions: PropTypes.shape({
-
-            /*
-             * Action creator to verify the user's email
-             */
-            verifyUserEmail: PropTypes.func.isRequired,
-
-            /*
-             * Action creator to update the user in the redux store
-             */
-            getMe: PropTypes.func.isRequired,
-            logError: PropTypes.func.isRequired,
-            clearErrors: PropTypes.func.isRequired,
-        }).isRequired,
-
-        /**
-         * Object reprenseting the current user
-         */
-        user: PropTypes.shape({
-            email_verified: PropTypes.bool,
-        }),
-
-        isLoggedIn: PropTypes.bool.isRequired,
-    }
+type Props = {
+    location: {
+        search: string;
+    };
+    siteName: string | undefined;
```

Comment:
You can use an optional parameter here as a shortcut for `string | undefined`:
```suggestion
    siteName?: string;
```